core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -353,12 +353,24 @@ public static Object getDataBasedOnDataTypeForNoDictionaryColumn(byte[] dataInBy
       } else if (actualDataType == DataTypes.STRING) {
         return getDataTypeConverter().convertFromByteToUTF8String(dataInBytes);
       } else if (actualDataType == DataTypes.SHORT) {
+        if (isEmptyByteArray(dataInBytes)) {
```

Comment:
Please add a comment .. for non string type no dictionary column empty byte array is null value so no  need to parse